### PR TITLE
Icon no dupes

### LINKED ISSUE #219: decorate icons leads to duplicated icons in sprite
### Expected Behaviour
`franklin-svg-sprite` should be a unique list of icons

### Actual Behaviour
`franklin-svg-sprite` contains duplicated icons, with duplicate ids, which I believe is not actually allowed because IDs must be unique

### Reproduce Scenario (including but not limited to)
pretty sure this would be reproducible on just about any project where more than one icon is used. can be see https://main--elixirsolutions--hlxsites.hlx.page where the sprite looks like:
```
<svg xmlns="http://www.w3.org/2000/svg" id="franklin-svg-sprite" style="display: none"><symbol id="icons-sprite-tag" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!-- Font Awesome Pro 5.15.4 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) --><path d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"></path></symbol><symbol id="icons-sprite-tag" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!-- Font Awesome Pro 5.15.4 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) --><path d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"></path></symbol><symbol id="icons-sprite-tag" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 640 512"><!-- Font Awesome Pro 5.15.4 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license (Commercial License) --><path d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"></path></symbol>
<symbol id="icons-sprite-search" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
  <path d="M16.9,15.5c2.4-3.2,2.2-7.7-0.7-10.6c-3.1-3.1-8.1-3.1-11.3,0c-3.1,3.2-3.1,8.3,0,11.4
    c2.9,2.9,7.5,3.1,10.6,0.6c0,0.1,0,0.1,0,0.1l4.2,4.2c0.5,0.4,1.1,0.4,1.5,0c0.4-0.4,0.4-1,0-1.4L16.9,15.5
    C16.9,15.5,16.9,15.5,16.9,15.5L16.9,15.5z M14.8,6.3c2.3,2.3,2.3,6.1,0,8.5c-2.3,2.3-6.1,2.3-8.5,0C4,12.5,4,8.7,6.3,6.3
    C8.7,4,12.5,4,14.8,6.3z"></path>
</symbol>
</svg>
```

note that `id="icons-sprite-tag"` is there 3 times

I believe the issue is  that https://github.com/adobe/helix-project-boilerplate/blob/main/scripts/lib-franklin.js#L180 appends to symbols rather than replacing it. 

In some quick local testing, just changing that to an assignment worked, but maybe there are cases where you need to do an append (I'm thinking multiple icons being decorated at the same time in asyc/await code?), in which case we might need to add icons individually only after checking if they already exist?

cc: @ramboz who I think set this up originally

### PR DESCRIPTION
**Pull request by shsteimer**
Fix #219 

I think fully resting this against a set of pages with multiple icons on it would be good, but haven't yet set that up.

Also probably can't be merged as is due to fstab change for my google drive. I can revert that, but would make setting up test pages with multiple icons difficult because I don't have edit access to https://drive.google.com/drive/folders/1MGzOt7ubUh3gu7zhZIPb7R7dyRzG371j

Before: https://main--franklin-playground--shsteimer.hlx.live
After: https://icon-no-dupes--franklin-playground--shsteimer.hlx.live

### CODE REVIEW

**Review by ramboz — APPROVED** [review]

**Comment by arumsey** [comment]
I tested this change against my pages that have several fragments with the same icon in them and only one icon sprite is now added.

File: scripts/lib-franklin.js [hunk 1]
```diff
@@ -176,7 +176,12 @@ export async function decorateIcons(element) {
     }
   }));
 
-  const symbols = Object.values(ICONS_CACHE).filter((v) => !v.styled).map((v) => v.html).join('\n');
+  const symbols = Object
+    .keys(ICONS_CACHE).filter((k) => !svgSprite.querySelector(`#icons-sprite-${k}`))
```
Comment (arumsey) [inline scripts/lib-franklin.js:180]: Why is the prefix `icons-sprite-` (plural) and not `icon-sprite-` (singular)?

**Comment by shsteimer** [comment]
@ramboz @davidnuescheler I do not have permissions/ability to merge this myself. What is the general process once PR is approved to merge?

**Comment by ramboz** [comment]
@shsteimer Sorry, was on PTO and missed the notification. Thanks for the contribution!

**Pull request merged**